Docker setup (#1241)
ElektraInitiative/PermaplanT: doc/development/01docker+local.md

# Docker + local

Run the database in Docker and the frontend and backend locally.

## Install Docker

[Install Docker](https://docs.docker.com/engine/install/) for your host OS.

## Installing Node 20 + Npm

> Make sure node 20 is installed, you may use to easly install multiple versions of node [nvm](https://github.com/nvm-sh/nvm),

Reviewer: Try to not have several mentions of which versions to install, this is hard to maintain. (Add link to where the version is already mentioned).

```suggestion
Make sure node, in the version as required for PermaplanT, is installed, you may use to easily install multiple versions of node [nvm](https://github.com/nvm-sh/nvm),
```

> Make sure node 20 is installed, you may use to easly install multiple versions of node [nvm](https://github.com/nvm-sh/nvm),
> use apt or [download node directly](https://nodejs.org/en/download).

Reviewer 2: ```suggestion
Make sure node 20 is installed, you may use [nvm](https://github.com/nvm-sh/nvm) to easily install multiple versions of node, use apt or [download node directly](https://nodejs.org/en/download).
```

## Install Rust

> Follow the sets to [install Rust using rustup](https://www.rust-lang.org/tools/install).

Reviewer: Again: link to which version we use.

Reviewer: @horenso Unfortunately, this does not always work. Sometimes a new version emits a warning, old ones didn't etc. Actually, the last upgrade to the latest stable rust version involved quite some effort.

As long as someone takes care that earlier and also the last version are always working, it would be okay. But the much easier variant is to simply support exactly one version and everyone is using this.

## Run the database

This runs the Postgis database and PgAdmin.

```bash

> cd docker

Reviewer: ```suggestion
cd .docker
```

Reviewer: @horenso simply add/commit suggestion if it is okay.
